DeleteCFStack/index.js

```diff
@@ -1,64 +1,75 @@
 const AWS = require('aws-sdk');
 const cf = new AWS.CloudFormation();
 
-async function handleWebhookEvent(webhookBody) {
-    const action = webhookBody['action'];
-    const pullRequest = webhookBody['pull_request'];
-    let message = 'Event ignored';
+function parseWebhookBody(eventType, webhookBody) {
+  let shouldDeleteStack, branchName, repoName;
 
-    // If no pull_request, ignore event
-    if(!pullRequest) {
-        return { statusCode: 200, message: message };
-    }
+  if (eventType === 'pull_request') {
+    branchName = webhookBody['pull_request']['head']['ref'];
+    repoName = webhookBody['pull_request']['head']['repo']['name'];
+    shouldDeleteStack = webhookBody['action'] === 'closed' && branchName !== 'master';
+  } else if (eventType === 'delete') {
+    branchName = webhookBody['ref'];
+    repoName = webhookBody['repository']['name'];
+    shouldDeleteStack = webhookBody['ref_type'] === 'branch' && branchName !== 'master';
+  };
 
-    const branchName = pullRequest['head']['ref'];
-    const repoName = pullRequest['head']['repo']['name'];
+  console.log('shouldDeleteStack', shouldDeleteStack);
+  const stackName = `${branchName}-${repoName}`
+  return { shouldDeleteStack, stackName }
+}
+
+async function handleWebhookEvent(eventType, webhookBody) {
+  if(!['pull_request', 'delete'].includes(eventType)) {
+    const message = `Event ignored. Event type: ${eventType}`;
+    return { statusCode: 200, body: message };
+  }
 
-    console.log({action});
-    console.log({branchName});
+  const { shouldDeleteStack, stackName } = parseWebhookBody(eventType, webhookBody);
 
-    if(action === 'closed' && branchName != 'master') {
-        const stackNameRepo = `${branchName}-${repoName}`;
-        const stackNameDev = `${stackNameRepo}-dev`;
-        const stackNameTest = `${stackNameRepo}-test`;
-        // Delete the stacks
-        await cf.deleteStack({StackName: stackNameRepo}).promise();
-        await cf.deleteStack({StackName: stackNameDev}).promise();
-        await cf.deleteStack({StackName: stackNameTest}).promise();
-        message = `CloudFormation Stacks deleted: ${stackNameRepo}, ${stackNameDev}, ${stackNameTest}`;
-    } else {
-        message = "Pull request event ignored";
-    }
+  if (!shouldDeleteStack) {
+    const message = `Event should not delete stack: shouldDeleteStack: ${shouldDeleteStack}, eventType: ${eventType}`;
     return { statusCode: 200, body: message };
-}
+  } else {
+    // Delete the stacks
+    const stackNameDev = `${stackName}-dev`;
+    const stackNameTest = `${stackName}-test`;
+    await cf.deleteStack({StackName: stackName}).promise();
+    await cf.deleteStack({StackName: stackNameDev}).promise();
+    await cf.deleteStack({StackName: stackNameDev}).promise();
```

Comment:
sorry I'm late to looking at this. is this the code that was merged? we're deleting `stackNameDev` twice here instead of `stackNameTest`